crates/noirc_evaluator/src/ssa/acir_gen/constraints.rs

```diff
@@ -395,19 +398,25 @@ pub(crate) fn to_radix_base(
     lhs: &Expression,
     radix: u32,
     limb_size: u32,
+    endianness: Endian,
     evaluator: &mut Evaluator,
 ) -> Vec<Witness> {
     // ensure there is no overflow
     let mut max = BigUint::from(radix);
     max = max.pow(limb_size) - BigUint::one();
     assert!(max < FieldElement::modulus());
 
-    let (result, bytes) = to_radix(radix, limb_size, evaluator);
+    let (mut result, bytes) = to_radix_little(radix, limb_size, evaluator);
+
+    if endianness == Endian::Big {
+        result.reverse();
+    }
+
     evaluator.opcodes.push(AcirOpcode::Directive(Directive::ToRadix {
         a: lhs.clone(),
         b: result.clone(),
         radix,
-        is_little_endian: true,
+        is_little_endian: endianness == Endian::Little,
```

Comment:
It should be: `is_little_endian: true,`

Reply:
It depends what witness you provide to the directive. You should provide the little_endian witness to the directive (with is_little_endian set to true), and only then reverse them for returning the result.

That way the directive is always used with little endian and we will be able to remove this parameter which leads to a more simple ACIR.

Reply:
Yes exactly, so you can call the directive with is_little_endian to true.

Reply:
Thanks for asking!
So no, unless I missed some point, it should work with is_little_endian set to true, if you do
```
 if endianness == Endian::Big {
         result.reverse();
     }
```
**after** the directive.